### ISSUE TITLE
Strange behavior of the Change Detection inside electron for IPC events

### ISSUE THREAD

**Issue by JGantner** [angular/angular#7154]
Today a ran into an issue with the change detection of Angular 2 beta6. 
The TLDR is that i needed to do a setInterval() call with an empty function to get angular to pick up changes in the component. 

I created a sample project that recreates the behavior. It can be found here: https://github.com/JGantner/angular2_change_detection_issue

Long Story:
I'm currently working on a Desktop that works with USB encryption hardware. To display the user that an encryption device is present i wanted to display an icon in the interface. So i created a component that listens for events from electrons node process sent via an electron ipc channel to the browser.

In the sample code to this issue the event is sent in the app/index.js line 34, 5 seconds after the main window is created. The event listener is located in the security-indicator-component (browser/security-level-indicator-component.ts line 16-20). In case you are wondering where the global ipcRenderer is defined. It is created in the browser/index.html file line 13. This was my workaround to get a reference to the ipcRenderer because system.js replaces the require function and i was not able to get it to load the electron stuff.

I'am wondering if there is a more elegant way that does not require the setInterval call in the SecurityLevelIndicator component.

Any help appreciated.

**Comment by robwormald** [comment]
This is because Zones (which trigger change detection) in Angular2 don't know anything about electron's IPC methods and thus they don't know to trigger change detection when an event arrives over IPC. setTimeout fixes it because Zones *does* know about that and so it triggers a change detection check.

See https://github.com/angular/angular/issues/5979 for an easy fix. See also https://github.com/angular/angular-electron/issues/7 - we should provide a simple service for this.
